pypdf/_page.py

```diff
@@ -956,6 +960,10 @@ def _merge_page_writer(
                         + cast(tuple, trsf.apply_on((q[4], q[5]), True))
                         + cast(tuple, trsf.apply_on((q[6], q[7]), True))
                     )
+                try:
+                    aa["/Popup"][NameObject("/Parent")] = aa.indirect_reference
```

Comment:
Why specifically /Popup? Could there be other annotations that need something similar?

Reply:
You can get a recursive call during cloning (with the force_duplicate) here I  try set back the parameter here

Reply:
This part is still something I stumble over. For example, why isn't the `FileAttachment` annotation or the `Link` annotation receiving the same treatment?

Reply:
the objective is to prevent recursive call if an object call itself. the only case I've detected is for popup pointed in aa["/Popup"] and where there is a possible attribute pointing back to aa in Popup["/Parent"]